src/data_processing/netflix_to_wikidata.py

```diff
@@ -72,7 +72,7 @@ def get_wikidata_data(title, year):
 
     wikidata_id = data['results']['bindings'][0]['item']['value'].split('/')[-1]
 
-    genres = [d['genreLabel']['value'] for d in data['results']['bindings'] if 'genreLabel' in d]
+    genres = {d['genreLabel']['value'] for d in data['results']['bindings'] if 'genreLabel' in d}
```

Comment:
Did you just use a "set" comprehension? I didn't realize such a thing existed!

Reply:
Yea I was trying to just eliminate duplicates with the least amount of effort, so just tried to make it a set and found that and it works. Though on second thought I'm not sure if there should be duplicates (if its getting data from other movies or something), so not 100% sure if it's a problem that necessarily should be solved.

Reply:
Makes sense. I think you should "manually" run the query on query.wikidata.org for any movies that are returning duplicate genres. Like you said, it could be that it's getting multiple movies. Or something else could be happening with the query that we need to debug. If you can gather the queries (or movie names/release dates) of such instances, we can debug this Thursday.

Reply:
I tested it for [Dragonheart](https://query.wikidata.org/#%20%20%20%20SELECT%20%2a%20WHERE%20%7B%0A%20%20%20%20%20%20%20%20SERVICE%20wikibase%3Amwapi%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20bd%3AserviceParam%20wikibase%3Aapi%20%22EntitySearch%22%20%3B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20wikibase%3Aendpoint%20%22www.wikidata.org%22%20%3B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20mwapi%3Asearch%20%22Dragonheart%22%20%3B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20wikibase%3Alimit%201%20%3B%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20mwapi%3Alanguage%20%22en%22%20.%0A%20%20%20%20%20%20%20%20%20%20%20%20%3Fitem%20wikibase%3AapiOutputItem%20mwapi%3Aitem%20.%0A%20%20%20%20%20%20%20%20%7D%0A%0A%20%20%20%20%20%20%20%20%3Fitem%20wdt%3AP31%2Fwdt%3AP279%2a%20wd%3AQ11424%20.%0A%0A%20%20%20%20%20%20%20%20%3Fitem%20wdt%3AP577%20%3FreleaseDate%20.%0A%20%20%20%20%20%20%20%20FILTER%20%28YEAR%28%3FreleaseDate%29%20%3E%3D%201995%20%26%26%20YEAR%28%3FreleaseDate%29%20%3C%3D%201997%29%20.%0A%20%20%20%20%20%20%20%20%0A%20%20%20%20%20%20%20%20OPTIONAL%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%3Fitem%20wdt%3AP136%20%3Fgenre%20.%0A%20%20%20%20%20%20%20%20%20%20%20%20%3Fgenre%20rdfs%3Alabel%20%3FgenreLabel%20.%0A%20%20%20%20%20%20%20%20%20%20%20%20FILTER%20%28lang%28%3FgenreLabel%29%20%3D%20%22en%22%29%20.%0A%20%20%20%20%20%20%20%20%7D%0A%0A%20%20%20%20%20%20%20%20%3Fitem%20rdfs%3Alabel%20%3FitemLabel%20.%0A%20%20%20%20%20%20%20%20FILTER%20%28lang%28%3FitemLabel%29%20%3D%20%22en%22%29%20.%0A%0A%20%20%20%20%20%20%20%20SERVICE%20wikibase%3Alabel%20%7B%20bd%3AserviceParam%20wikibase%3Alanguage%20%22en%22%20.%20%7D%0A%20%20%20%20%7D%0A%20%20%20%20LIMIT%2010%0A) and it appears that there are multiple release dates in different countries which is causing it to be returned 5 times. So there is at least one scenario where it is returning a valid movie but still includes duplicates. Maybe could just change it to only search for US release date or something? I'll look at the other ones to see if there is anything else causing it.

Reply:
Other duplicates in the first 100 include:
Presumably caused by multiple release dates -
[Mostly Martha](https://www.wikidata.org/wiki/Q684936)
[Silkwood](https://www.wikidata.org/wiki/Q1412037)
[The Killing](https://www.wikidata.org/wiki/Q592072)
[Jingle all the Way](https://www.wikidata.org/wiki/Q1064530)
[Congo](https://www.wikidata.org/wiki/Q598860)
[At Home Among Strangers](https://www.wikidata.org/wiki/Q1963834)
[Richard III](https://www.wikidata.org/wiki/Q2501468)
[Jade](https://www.wikidata.org/wiki/Q889386)
[The Spitfire Grill](https://www.wikidata.org/wiki/Q1212954)
Valid movie, unsure what's causing it - 
The classic [Powerpuff Girls Movie](https://www.wikidata.org/wiki/Q1199301)
Film and television film within same year -
The Love Letter [Film](https://www.wikidata.org/wiki/Q949305), [Television Film](https://www.wikidata.org/wiki/Q3363005)

Reply:
So yeah basically it seems like the most is due to the release dates issue. Which could probably be solved more efficiently, but technically valid to just remove duplicates.
2 are because of the date range issue. I'll try running it with just the single date and see how much that changes the results.
2 are unknown reasons.

Reply:
Ok, I changed it to just filter for the exact year instead of the range, which resolves the last 2 (though the first one still has 2 release dates). It only misses out on 2 movies (53 instead of 55), which is probably worth the improvement in the accuracy of the data. So yeah, now basically all of them are caused by multiple release dates instead of Powerpuff Girls and Rudolph, not sure what is causing those.

Reply:
It took a bit of wrangling, but here is the query to filter by US release date: https://w.wiki/B6av

The key is:

```
        ?item p:P577 ?releaseDateStatement .
        ?releaseDateStatement ps:P577 ?releaseDate .
        ?releaseDateStatement pq:P291 wd:Q30 .
```

Reply:
So apparently only 10% have a US release date
I tried to update it to get either US release date or unspecified which increases it to ~51%